docs/ConfigurationsViaEnvironmentVariables.md

```diff
@@ -0,0 +1,293 @@
+
+In order to overcome the above, configurations can be provided via environment variables or system properties following a specific pattern.
+
+## Environment variable / system property pattern
+
+`<NAMESPACE>_<CONFIGURATION>="<VALUE>"`
+
+![Environment variable / system property structure](images/ConfigurationsViaEnvironmentVariables/VarStructure.png)
+
+#### Example 1:
+
+Consider the below YAML configuration.
+
+```yaml
+wso2.configuration:
+  complexBean:
+    name: default
+    Bean:
+      name: default
+```
+
+The above configuration can be represented with the below environment variables / system properties:
+
+```bash
+WSO2.CONFIGURATION_COMPLEXTESTBEAN_NAME="default"
+WSO2.CONFIGURATION_COMPLEXTESTBEAN_BEAN_NAME="default"
+```
+
+Where,
+
+* NAMESPACE = `wso2.configuration`
+
+___
+**Notes:** 
+
+* As a convention, 
+  * The environment variable names / system property names should be in all caps.
+  * The environment variable / system property segments should be separated by an underscore ("_").
+___
+
+#### Example 2:
+
+Consider the below YAML configuration.
+
+```yaml
+testconfiguration:
+  tenant: tenant
+  transports:
+    transport:
+      -  name: abc
+         port: 8000
+         secure: false
+         desc: This transport will use 8000 as its port
+         password: password
+      -  name: pqr
+         port: 8001
+         secure: true
+         desc: This transport will use 8001 as its port. Secure - true
+      -  name: xyz
+         port: 9000
+         secure: true
+         desc: This transport will use 9000 as its port
+```
+
+The above configuration can be represented with the below environment variables / system properties:
+
+```bash
+TESTCONFIGURATION_TENANT="tenant"
+
+# Transport "abc"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_NAME="abc"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_PORT="8000"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_SECURE="false"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_DESC="This transport will use 8000 as its port"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_0_PASSWORD="password"
+
+# Transport "pqr"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_NAME="pqr"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_PORT="8001"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_SECURE="true"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_1_DESC="This transport will use 8001 as its port. Secure - true"
+
+# Transport "xyz"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_NAME="xyz"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_PORT="9000"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_SECURE="true"
+TESTCONFIGURATION_TRANSPORTS_TRANSPORT_2_DESC="This transport will use 9000 as its port"
+```
+
+Where,
+
+* NAMESPACE = `testconfiguration`
+
+___
+**Notes:** 
+
+* As a convention, 
+  * The environment variable names / system property names should be in all caps.
+  * The environment variable / system property segments should be separated by an underscore ("_").
```

Comment:
Maybe we could include a sample showing what's correct and what's not?

Reply:
Included a reference to the section in line 110